src/server/server_impl.go

```diff
@@ -47,20 +51,32 @@ func (server *server) Start() {
 	go func() {
 		addr := fmt.Sprintf(":%d", server.debugPort)
 		logger.Warnf("Listening for debug on '%s'", addr)
-		logger.Info(http.ListenAndServe(addr, server.debugListener.debugMux))
+		list, err := reuseport.Listen("tcp", addr)
```

Comment:
You'll need to close the debug listener when the graceful stop is called. So the listener will need to live as a field on the Server.